astore/client/BUILD.bazel

```diff
@@ -31,15 +31,6 @@ go_binary(
     visibility = ["//visibility:public"],
 )
 
```

Comment:
Remove the macOS target because it now causes a build conflict with the cross-compiled enkit target for windows and linux. The linux enkit binary also natively works on ARM instruction set for macOS.

Reply:
> Remove the macOS target because it now causes a build conflict with the cross-compiled enkit target for windows and linux

what is the build conflict?

> The linux enkit binary also natively works on ARM instruction set for macOS.

There's more to it than instruction set... this seems fishy/coincidental

Reply:
This is the error message. Note the ` -installsuffix darwin_amd64`:

```
ERROR: /home/bbhuynh/enkit/lib/kcerts/BUILD.bazel:3:11: GoCompilePkg lib/kcerts/kcerts.a failed: (Exit 1): process-wrapper failed: error executing command 
  (cd /home/bbhuynh/.cache/bazel/_bazel_bbhuynh/ccbc8eec90d71932e4b174477612b27d/sandbox/processwrapper-sandbox/5089/execroot/enkit && \
  exec env - \
    CGO_ENABLED=0 \
    GOARCH=amd64 \
    GOEXPERIMENT=nocoverageredesign \
    GOOS=darwin \
    GOPATH='' \
    GOROOT=bazel-out/k8-fastbuild-ST-905cad94ae44/bin/external/io_bazel_rules_go/stdlib_ \
    GOROOT_FINAL=GOROOT \
    TMPDIR=/tmp \
  /home/bbhuynh/.cache/bazel/_bazel_bbhuynh/install/50a5a3eaeaa5297d31568ab6a3a702d9/process-wrapper '--timeout=0' '--kill_delay=15' '--stats=/home/bbhuynh/.cache/bazel/_bazel_bbhuynh/ccbc8eec90d71932e4b174477612b27d/sandbox/processwrapper-sandbox/5089/stats.out' bazel-out/k8-opt-exec-2B5CBBC6-ST-98c4feb8e515/bin/external/go_sdk_1_20/builder_reset/builder compilepkg -sdk external/go_sdk_1_20 -installsuffix darwin_amd64 -src lib/kcerts/cache.go -src lib/kcerts/certs.go -src lib/kcerts/keys.go -src lib/kcerts/signer.go -src lib/kcerts/ssh.go -src lib/kcerts/ssh_linux.go -src lib/kcerts/ssh_windows.go -src lib/kcerts/types.go -embedroot '' -embedroot bazel-out/k8-fastbuild-ST-2c2713b264f6/bin -embedlookupdir lib/kcerts -arc 'github.com/enfabrica/enkit/lib/cache=github.com/enfabrica/enkit/lib/cache=bazel-out/k8-fastbuild-ST-2c2713b264f6/bin/lib/cache/cache.x' -arc 'github.com/enfabrica/enkit/lib/config/directory=github.com/enfabrica/enkit/lib/config/directory=bazel-out/k8-fastbuild-ST-2c2713b264f6/bin/lib/config/directory/directory.x' -arc 'github.com/enfabrica/enkit/lib/config/marshal=github.com/enfabrica/enkit/lib/config/marshal=bazel-out/k8-fastbuild-ST-2c2713b264f6/bin/lib/config/marshal/marshal.x' -arc 'github.com/enfabrica/enkit/lib/kcerts/ked25519=github.com/enfabrica/enkit/lib/kcerts/ked25519=bazel-out/k8-fastbuild-ST-2c2713b264f6/bin/lib/kcerts/ked25519/ked25519.x' -arc 'github.com/enfabrica/enkit/lib/kflags=github.com/enfabrica/enkit/lib/kflags=bazel-out/k8-fastbuild-ST-2c2713b264f6/bin/lib/kflags/kflags.x' -arc 'github.com/enfabrica/enkit/lib/logger=github.com/enfabrica/enkit/lib/logger=bazel-out/k8-fastbuild-ST-2c2713b264f6/bin/lib/logger/logger.x' -arc 'github.com/enfabrica/enkit/lib/multierror=github.com/enfabrica/enkit/lib/multierror=bazel-out/k8-fastbuild-ST-2c2713b264f6/bin/lib/multierror/multierror.x' -arc 'github.com/enfabrica/enkit/lib/srand=github.com/enfabrica/enkit/lib/srand=bazel-out/k8-fastbuild-ST-2c2713b264f6/bin/lib/srand/srand.x' -arc 'github.com/mitchellh/go-homedir=github.com/mitchellh/go-homedir=bazel-out/k8-fastbuild-ST-2c2713b264f6/bin/external/com_github_mitchellh_go_homedir/go-homedir.x' -arc 'golang.org/x/crypto/ssh=golang.org/x/crypto/ssh=bazel-out/k8-fastbuild-ST-2c2713b264f6/bin/external/org_golang_x_crypto/ssh/ssh.x' -arc 'golang.org/x/crypto/ssh/agent=golang.org/x/crypto/ssh/agent=bazel-out/k8-fastbuild-ST-2c2713b264f6/bin/external/org_golang_x_crypto/ssh/agent/agent.x' -importpath github.com/enfabrica/enkit/lib/kcerts -p github.com/enfabrica/enkit/lib/kcerts -package_list bazel-out/k8-opt-exec-2B5CBBC6-ST-98c4feb8e515/bin/external/go_sdk_1_20/packages.txt -o bazel-out/k8-fastbuild-ST-2c2713b264f6/bin/lib/kcerts/kcerts.a -x bazel-out/k8-fastbuild-ST-2c2713b264f6/bin/lib/kcerts/kcerts.x -gcflags '' -asmflags '')
lib/kcerts/ssh.go:231:15: undefined: DialTimeout
lib/kcerts/ssh.go:298:15: undefined: DialTimeout
lib/kcerts/ssh.go:332:15: undefined: DialTimeout
compilepkg: error running subcommand external/go_sdk_1_20/pkg/tool/linux_amd64/compile: exit status 2
```

Reply:
Correction: I fixed the darwin (mac) build issue by creating `ssh_darwin.go` that is compiled into the binary when `GOOS=darwin` is set by bazel as a build flag. The amd64-mac binary is compatible with the arm64-mac platform after I built the `enkit` binary for amd64-mac and ran a bunch of `enkit` commands on a Macbook Air.

Reply:
Yep! Adding a mac OS compatible variant is a better fix IMO because we don't need to make the decision right now to drop support for mac OS.